File: majority-element/majority-element.js
// Moore's voting algorithm

var majorityElement = function (nums) {
    majority = nums[0];
    count = 1;

    for (let i = 0; i < nums.length; i++) {
        if (majority == nums[i]) {
            count += 1;
        } else {
            count -= 1;
        }


        if (count == 0) {
            majority = nums[i];
            count = 1;
        }
    }

    return majority;


};

File: majority-element/majority-element.py
# Moore's Voting Algorithm

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        majority = nums[0]
        count = 1 
        
        for i in nums:
            if i == majority:
                count += 1 
            else:
                count -= 1
                
            if count == 0:
                majority = i 
                count = 1 
                
                
        return majority 